pandas/io/formats/excel.py

```diff
@@ -85,10 +89,10 @@ def __init__(
         **kwargs,
     ) -> None:
         if css_styles and css_converter:
-            css = ";".join(
-                [a + ":" + str(v) for (a, v) in css_styles[css_row, css_col]]
-            )
-            style = css_converter(css)
+            # Use dict to get only one declaration per property, then convert
+            # to frozenset for caching
+            unique_declarations = frozenset(dict(css_styles[css_row, css_col]).items())
```

Comment:
Maybe safer to use a tuple instead of a frozenset to guarantee the ordering for a consistent cache key?

Reply:
I think we actually want it to be order-invariant here. Here's my reasoning:

1. The `dict(...)` ensures that each unique property is only present once; specifically, the final declaration in the CSS string for each property is the one that is stored in the dict.
2. The `.items()` call converts that dictionary into `(property, value)` pairs. By (1), this collection of pairs is guaranteed to have each property show up only once.
3. The frozenset has the same hash iff every tuple in the collection of pairs is the same, with no concept of order. Since the set can't contain duplicate properties, then two sets with the same hash must represent the same declarations. For example, if in Cell A1 I applied `font-color: red; font-weight: bold` and Cell B1 I applied `font-weight: bold; font-color: red`, they can still be safely cached as one.

I added another set of commits that added tests to ensure it's working as intended. Please take a look. If my logic isn't sound or I'm overlooking any edge cases, please let me know!